src/devices/MotorHat/MotorPinProvider.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+
+namespace Iot.Device.MotorHat
+{
+    /// <summary>
+    /// Represents a provider for <see cref="MotorPins"/>
+    /// </summary>
+    public interface IMotorPinProvider
+    {
+        /// <summary>
+        /// Gets the <see cref="MotorPins"/> for motor at the specified <paramref name="index"/>
+        /// </summary>
+        /// <param name="index">The index of the motor to get pins for</param>
+        /// <returns></returns>
+        public abstract MotorPins GetPinsForMotor(int index);
+    }
+
+    /// <summary>
+    /// Abstract base class for creating DCMotor instances
```

Comment:
Documentation is missleading. This is not an abstract class.

Reply:
Updated.